File: iosApp/iosApp/NativeViewFactory.swift
//
//  NativeViewFactory.swift
//  iosApp
//
//  Created by Nataliia Mantrova on 26.04.2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import SwiftUI
import ComposeApp


class IOSNativeViewFactory: NativeViewFactory {
    static var shared = IOSNativeViewFactory()
    
    func createDatePickerView(
        onDateSelect: @escaping (KotlinLong) -> Void,
        onDismiss: @escaping () -> Void
    ) -> UIViewController {
        let view = DatePickerView(onDateSelect: onDateSelect, onDismiss: onDismiss)
        return UIHostingController(rootView: view)
    }
    
    func createProvideTextView(
        text: String,
        onDone: @escaping (String) -> Void,
        onDismiss: @escaping () -> Void
    ) -> UIViewController {
        let view = ProvideTextView(text: text, onDone: onDone, onDismiss: onDismiss)
        return UIHostingController(rootView: view)
    }
}

File: composeApp/src/iosMain/kotlin/com/testproject/presentation/NativeViewFactory.kt
package com.testproject.presentation

import platform.UIKit.UIViewController

interface NativeViewFactory {
    fun createDatePickerView(
        onDateSelect: (Long) -> Unit,
        onDismiss: () -> Unit,
    ): UIViewController

    fun createProvideTextView(
        text: String,
        onDone: (String) -> Unit,
        onDismiss: () -> Unit,
    ): UIViewController
}